taca/storage/storage.py

```diff
@@ -44,23 +51,30 @@ def archive_to_swestore(config, days, run=None):
     # Otherwise find all runs in every data dir on the nosync partition
     else:
         LOG.info("Archiving old runs to SWESTORE")
-        for data_dir in config.get('storage').get('data_dirs'):
-            to_send_dir = os.path.join(data_dir, 'nosync')
+        for to_send_dir in config.get('storage').get('archive_dirs'):
             LOG.info('Checking {} directory'.format(to_send_dir))
             with filesystem.chdir(to_send_dir):
-                for run in [r for r in os.listdir(to_send_dir) if re.match(filesystem.RUN_RE, r)]:
-                    _archive_run(config, run)
-
+                to_be_archived = [r for r in os.listdir(to_send_dir) if re.match(filesystem.RUN_RE, r)
+                                            and not os.path.exists("{}.archiving".format(r.split('.')[0]))]
+                if to_be_archived:
+                    pool = Pool(processes=len(to_be_archived))
+                    pool.map_async(_archive_run, ((run,) for run in to_be_archived))
+                    pool.close()
+                    pool.join()
```

Comment:
It will basically throw the 6 processes and the 6 processes will keep competing for resources. Basically they will keep alternating. This is inefficient of course. We could introduce a more complex logic to avoid that, but _I_ don't think its worth it, as you say, it is very unlikely to have 24 runs to archive, plus the CPU spike on archiving is very short, the main waiting time is waiting for swestore to perform the checksums.